tests/evm_backends/revm_env.py

```diff
@@ -0,0 +1,124 @@
+import re
+from contextlib import contextmanager
+
+from eth_keys.datatypes import PrivateKey
+from pyrevm import EVM, BlockEnv, Env
+
+from tests.evm_backends.base_env import BaseEnv, EvmError, ExecutionResult
+
+
+class RevmEnv(BaseEnv):
+    INVALID_OPCODE_ERROR = "InvalidFEOpcode"
+
+    def __init__(
+        self,
+        gas_limit: int,
+        account_keys: list[PrivateKey],
+        tracing=False,
+        block_number=1,
+        evm_version="latest",
+    ) -> None:
+        super().__init__(gas_limit, account_keys)
+        self._evm = EVM(
+            gas_limit=gas_limit,
+            tracing=tracing,
+            spec_id=evm_version,
+            env=Env(block=BlockEnv(number=block_number)),
+        )
+
+    @contextmanager
+    def anchor(self):
+        snapshot_id = self._evm.snapshot()
+        try:
+            yield
+        finally:
+            try:
+                self._evm.revert(snapshot_id)
+            except OverflowError:
+                # snapshot_id is reverted by the transaction already.
+                # revm updates are needed to make the journal more robust.
+                pass
+
+    def get_balance(self, address: str) -> int:
+        return self._evm.get_balance(address)
+
+    def set_balance(self, address: str, value: int):
+        self._evm.set_balance(address, value)
+
+    @property
+    def accounts(self) -> list[str]:
+        return [key.public_key.to_checksum_address() for key in self._keys]
+
+    @property
+    def block_number(self) -> int:
+        return self._evm.env.block.number
+
+    @property
+    def timestamp(self) -> int | None:
+        return self._evm.env.block.timestamp
+
+    @property
+    def last_result(self) -> ExecutionResult:
+        result = self._evm.result
+        return ExecutionResult(
+            gas_refunded=result.gas_refunded,
+            gas_used=result.gas_used,
+            is_success=result.is_success,
+            logs=result.logs,
+        )
+
+    def execute_code(
+        self,
+        to: str,
+        sender: str | None = None,
+        data: bytes | str = b"",
+        value: int = 0,
+        gas: int | None = None,
+        gas_price: int = 0,
+        is_modifying: bool = True,
+    ):
+        data = data if isinstance(data, bytes) else bytes.fromhex(data.removeprefix("0x"))
+        try:
+            output = self._evm.message_call(
+                to=to,
+                caller=sender or self.deployer,
+                calldata=data,
+                value=value,
+                gas=self.gas_limit if gas is None else gas,
+                gas_price=gas_price,
+                is_static=not is_modifying,
+            )
+            return bytes(output)
```

Comment:
might be helpful to add a note, what the original type of `output` is

Reply:
pyrevm was sending list[u8] but I fixed it upstream. This line is not necessary anymore